pkg/controller/jobframework/reconciler.go

```diff
@@ -104,14 +104,33 @@ func NewReconciler(
 
 func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
 	object := job.Object()
-	if err := r.client.Get(ctx, req.NamespacedName, object); err != nil {
-		// we'll ignore not-found errors, since there is nothing to do.
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+	log := ctrl.LoggerFrom(ctx).WithValues("job", req.String())
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	err := r.client.Get(ctx, req.NamespacedName, object)
+
+	if apierrors.IsNotFound(err) || !object.GetDeletionTimestamp().IsZero() {
```

Comment:
If the object is a child job and the job has matched this condition, I guess the below processes related to finalizers always fail.
wdyt?


Also, what happens if the object isn't managed by kueue? I guess L118 always happens.

Reply:
In any of the cases  
```
 r.client.List(ctx, &workloads, client.InNamespace(req.Namespace),
			client.MatchingFields{getOwnerKey(job.GetGVK()): req.Name})
```
should result in an empty list , the reconcile will stop just like before.

Reply:
My concern is that there is an error in the logs. I'm wondering how to handle logging in this case.

Reply:
When debugging kueue, users are confused by unrelated logs.

Reply:
What error?

Reply:
I meant the info log (L113). But maybe I'm too worried.

Reply:
removed